django/middleware/csrf.py

```diff
@@ -160,20 +161,49 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                try:
+                    csrf_token = request.session[settings.CSRF_SESSION_KEY]
+                except AttributeError:
+                    raise ImproperlyConfigured(
+                        'CSRF_USE_SESSIONS is enabled, but request.session is not set. Please make '
+                        'sure you have sessions set up correctly.'
+                    )
+            else:
+                cookie_token = request.COOKIES[settings.CSRF_COOKIE_NAME]
+                csrf_token = _sanitize_token(cookie_token)
+
+                if csrf_token != cookie_token:
+                    # Cookie token needed to be replaced;
+                    # the cookie needs to be reset.
+                    request.csrf_cookie_needs_reset = True
+            return csrf_token
+        except KeyError:
```

Comment:
I kinda dislike that `KeyError` spans this much code, but I do not see a nice way around it either. On a minor nitpick I'd also rename `_get_current_token` to just `_get_token` just to be in line with `_set_token`